pkg/auth/providers/azure/azure_actions.go

```diff
@@ -73,14 +73,29 @@ func (ap *azureProvider) testAndApply(actionName string, action *types.Action, r
 
 	azureADConfig := &azureADConfigApplyInput.Config
 
-	// This covers the case where users upgrade Rancher to v2.6.7+ without having used Azure AD as the auth provider.
-	// In 2.6.7+, whether Azure AD is later registered or not, Rancher on startup creates the annotation on the template auth config.
-	// But in the case where the auth config had been created on Rancher startup prior to v2.6.7, the annotation would be missing.
-	// This ensures the annotation is set on next auth provider setup attempt.
-	if azureADConfig.ObjectMeta.Annotations == nil {
-		azureADConfig.ObjectMeta.Annotations = make(map[string]string)
-	}
-	azureADConfig.ObjectMeta.Annotations[GraphEndpointMigratedAnnotation] = "true"
+	// Get the current config from the database and, if it is enabled and does not have the new auth flow annotation,
+	// do not set the annotation on the new config.
+	// This covers the case where admins had been using a deprecated Azure AD setup
+	// and must migrate to the new auth config setup after upgrading Rancher.
+	// But they may not want to migrate immediately, and could still want to reconfigure their existing, deprecated config by changing
+	// the Application Secret after rotating it, for example. This way, Rancher allows the old setup to be reconfigured.
+	currentConfig, err := ap.getAzureConfigK8s()
+	if err != nil {
+		return err
```

Comment:
I think you want to wrap this in an `httperror` so that we have more control over the response to the user.

Reply:
Errors returned directly from the action (except for login errors which are caught and handled differently) are just returned to the UI with a 500 response code. I tested this particular case, and the user would just see the error's string that is returned here.

Reply:
Thank you for testing that out. I don't think we want user to see the error string returned here. We would want a generic error message and the true error message to be logged server side. 

In short, I hold to what I said above - we should wrap this with a generic error message and a specific response code.

Reply:
Agreed, but, ideally, I will need to change all the other places where this method is called. Some of them are "user-facing" too.